FogCreek/Glitch-Community
# Pool rodent - Private Collections

## Links
* Remix link: https://pool-rodent.glitch.me/
* Issue-tracker link: this pr covers several tickets in the [Private Collections epic](https://app.zenhub.com/workspaces/community-5d07d1359dee6059a9688176/board?assignees=sarahzinger&epics=192398789_427&repos=117580951,192398789) including: 
    - https://app.zenhub.com/workspaces/community-5d07d1359dee6059a9688176/issues/fogcreek/glitch-community-work/431
    - https://app.zenhub.com/workspaces/community-5d07d1359dee6059a9688176/issues/fogcreek/glitch-community-work/433
    - https://app.zenhub.com/workspaces/community-5d07d1359dee6059a9688176/issues/fogcreek/glitch-community-work/432
* Specs/doc links: https://docs.google.com/document/d/1E-gJwFbmOUxwEAYLIGLNWFJwZsOXLSvBPhJuCN80UHA/edit#bookmark=id.xrq7m23ovvia

## GIF/Screenshots:
<img width="847" alt="Screen Shot 2019-09-17 at 11 26 47 AM" src="https://user-images.githubusercontent.com/6620164/65056036-357b8d00-d93e-11e9-8f56-d092ae6001d8.png">
<img width="319" alt="Screen Shot 2019-09-17 at 11 27 35 AM" src="https://user-images.githubusercontent.com/6620164/65056038-36142380-d93e-11e9-92a4-ae5d0bc3dfe6.png">
<img width="328" alt="Screen Shot 2019-09-17 at 11 27 47 AM" src="https://user-images.githubusercontent.com/6620164/65056039-36142380-d93e-11e9-893f-871b03f11b0d.png">
<img width="361" alt="Screen Shot 2019-09-17 at 11 14 38 AM" src="https://user-images.githubusercontent.com/6620164/65055509-4d064600-d93d-11e9-9df5-a3d4ef18172f.png">

## Changes:
Adds Private Collections to the frontend and updates:
- the collections list on a user page, to have a view only privacy badge on private collections
- the collection page itself to have a privacy toggle
- the add project to collections popover to make private collection collections yellow

## How To Test:
* unfortunately the backend for this has yet to be deployed: https://github.com/FogCreek/Glitch/pull/1924 so testing is tricky unless you know how to run the backend locally (which is very possible and I can attempt to remember how to set it up if you want i can try to show ya! but it's definitely a little bit of a pita right now), we may wish to wait until that's shipped before we ship this although it's not necessary because...
* I have written this in such a way that it should all be properly darkshipped behind the `My Stuff` feature flag so as long as you don't have that checked everything should look exactly as it used to be which is good to test.

## Feedback I'm looking for:
- anything I'm missing? any style bugs? One thing that occurs to me is I'm not sure it's clear to users who use a screenreader if private collections (or projects!) in an add project to collection popover are private, as I think the only thing we do is make the background yellow. Perhaps I should abstract this into `<Result Item />` as a private prop that changes the background and maybe adds a visually hidden word of "Private" the whatever the screenreader announces? I was thinking though that that might be something to do after we've incorporated the ResultItems from `shared-components` in a separate pr. Let me know your thoughts and I can make a card for that work if it belongs elsewhere, or if we want to handle it here I can update this pr directly :)
